File: backend/api-gateway/src/main/java/ru/app/apigateway/repository/UserRepository.java
package ru.app.apigateway.repository;

import reactor.core.publisher.Mono;
import ru.app.apigateway.entity.User;
import org.springframework.data.r2dbc.repository.R2dbcRepository;

public interface UserRepository extends R2dbcRepository<User, Long> {
    Mono<User> findByEmail(String email);
    Mono<User> findByUsername(String username);
}

File: backend/user-service/src/main/kotlin/ru/doedating/userservice/repository/UserRepository.kt
package ru.doedating.userservice.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.doedating.userservice.entity.User

interface UserRepository: JpaRepository<User, Long> {
    fun findByEmail(email: String): User?

    fun findByUsername(username: String): User?
}